ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/__init__.py

```diff
@@ -0,0 +1,165 @@
+
+import importlib
+from collections import ChainMap
+from functools import cached_property
+from ansible_collections.arista.avd.plugins.plugin_utils.avdfacts import AvdFacts
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get, AristaAvdError
+
+
+DEFAULT_AVD_INTERFACE_DESCRIPTIONS_PYTHON_MODULE = (
+    "ansible_collections.arista.avd.roles.eos_designs.python_modules.interface_descriptions"
+)
+DEFAULT_AVD_INTERFACE_DESCRIPTIONS_PYTHON_CLASS_NAME = "AvdInterfaceDescriptions"
+
+
+class AvdInterfaceDescriptions(AvdFacts):
+
+    '''
+    Class used to render Interface Descriptions either from custom Jinja2 templates or using default Python Logic
+
+    Since some templates might contain certain legacy variables (switch_*),
+    those are mapped from the switch.* model
+
+    This class is imported adhoc based on the variable `templates.interface_descriptions.python_module` so it can
+    be overridden by a custom python class.
+    '''
+
+    @cached_property
+    def _default_mpls_overlay_role(self) -> str:
+        return get(self._hostvars, "switch.default_mpls_overlay_role")
+
+    @cached_property
+    def _mpls_lsr(self) -> str:
+        return get(self._hostvars, "switch.mpls_lsr")
+
+    @cached_property
+    def _mlag_peer(self) -> str:
+        return get(self._hostvars, "switch.mlag_peer", required=True)
+
+    @cached_property
+    def _mlag_port_channel_id(self) -> str:
+        return get(self._hostvars, "switch.mlag_port_channel_id", required=True)
+
+    def underlay_ethernet_interfaces(self, link_type: str, link_peer: str, link_peer_interface: str) -> str:
+        if template_path := get(self._hostvars, "switch.interface_descriptions.underlay_ethernet_interfaces"):
+            template_vars = ChainMap({}, self._hostvars)
+            template_vars['link'] = {
+                "type": link_type,
+                "peer": link_peer,
+                "peer_interface": link_peer_interface,
+            }
+            return self.template_var(template_path, template_vars)
+
+        link_peer = str(link_peer).upper()
+        if link_type == "underlay_p2p":
+            return f"P2P_LINK_TO_{link_peer}_{link_peer_interface}"
```

Comment:
wondering if we would not be better using "_".join by default in our scenarios to make sure that even if a variable is None we don't end up with a weird `DESCRIPTION__PO_`. Though in theory I agree we "know" some of these variables can never be empty

Reply:
In this function these two vars are required inputs, so we know they are set. The purpose here is 1:1 refactor.